src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,50 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact information in the address book. Guarantees:
+ * immutable; is valid as declared in {@link #isValidAddress(String)}
+ */
+
+public class Contact {
+
+	public String value;
+	protected boolean isPrivate;
+
+	 /**
+     * Validates given contact information.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+	
+	public Contact(String address, boolean isPrivate, String contactTypeValidationRegex, String contactTypeConstraints) throws IllegalValueException {
```

Comment:
Interesting design by capturing the validation regex and messages in constructor.